{code}
Invalid event T_ATTEMPT_FAILED on Task task_1234_5678_1_01_000001
{code}
The task had more than one running attempts (because of speculative execution), while one of them succeeded and the task was marked succeeded, the second failed and caused the Task state machine to enter error state since the task was in a leaf vertex and does the following:
{code}
if (task.leafVertex) {
        LOG.error("Unexpected event for task of leaf vertex " + event.getType() + ", taskId: "
            + task.getTaskId());
        task.internalError(event.getType());
      }
{code}
This JIRA tracks fixing this invalid state.